frontends/web/src/routes/bitsurance/widget.tsx

```diff
@@ -99,18 +99,50 @@ export const BitsuranceWidget = ({ code }: TProps) => {
     current.contentWindow?.postMessage(message, '*');
   };
 
+  const sendAddressWithXPub = (address: string, sig: string, xpub: string) => {
+    const { current } = iframeRef;
+
+    if (!current) {
+      return;
+    }
+
+    const message = serializeMessage({
+      version: MessageVersion.V0,
+      type: V0MessageType.Address,
+      bitcoinAddress: address,
+      extendedPublicKey: xpub,
+      signature: sig,
+    });
+
+    current.contentWindow?.postMessage(message, '*');
+  };
+
   const handleRequestAddress = (message: RequestAddressV0Message) => {
     signing = true;
     const addressType = message.withScriptType ? String(message.withScriptType) : '';
     const withMessageSignature = message.withMessageSignature ? message.withMessageSignature : '';
+    const withExtendedPublicKey: boolean = message.withExtendedPublicKey ? message.withExtendedPublicKey : false;
     signAddress(
       addressType,
       withMessageSignature,
       code)
       .then(response => {
         signing = false;
         if (response.success) {
-          sendAddress(response.address, response.signature);
+          if (withExtendedPublicKey) {
+            const bitcoinSimple = accountInfo?.signingConfigurations[0].bitcoinSimple;
```

Comment:
drive-by comment: the wpkh xpub might not be in the first location - it's safer to look it up using the scriptType in case the order changes in the future

Reply:
Good idea. Will improve the lookup.